src/test/java/org/dataloader/DataLoaderStatsTest.java

```diff
@@ -63,8 +65,8 @@ public void stats_are_collected_by_default() {
     public void stats_are_collected_with_specified_collector() {
         // lets prime it with some numbers so we know its ours
         StatisticsCollector collector = new SimpleStatisticsCollector();
-        collector.incrementLoadCount();
-        collector.incrementBatchLoadCountBy(1);
+        collector.incrementLoadCount(new IncrementLoadCountStatisticsContext<>(1, null));
+        collector.incrementBatchLoadCountBy(1, new IncrementBatchLoadCountByStatisticsContext<>(1, null));
 
         BatchLoader<String, String> batchLoader = CompletableFuture::completedFuture;
         DataLoaderOptions loaderOptions = DataLoaderOptions.newOptions().setStatisticsCollector(() -> collector);
```

Comment:
I think we need a test such that we tets that give a `load(K,context)` in the data loader, we get `K` and `context` passed into the StatsCollector.  That is the innards of DataLoader do what we expect them to do